students/itsabhi/Lesson7/html_render.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python3
+
+
+class Element:
+
+    tag = ""
+    indent = "   "
+
+    def __init__(self, content=None, **kwargs):
+        # Takes optional keyword arguments that can be used to initialize additional element parameters like font etc.
+        if isinstance(content, str):
+            # Initializes a list with the content
+            self.content = [str(content)]
+            for key in kwargs:
+                self.another_attrib_key = key
+                self.another_attrib_value = kwargs[key]
+        else:
+            self.content = []
+
+    def append(self, add_content):
+        self.content.append(add_content)
+
+    def render(self, file_out, cur_ind=""):
+        # Goes through the content list and renders each list item. If the item is an Element object,
+        # then it does recursive call to render method, else for string object it writes to file.
+
+        # Uncomment below line for Step 1
+        # self.tag = "html"
+        f = file_out
+        if hasattr(self, "another_attrib_value"):
+            f.write(self.opening_tag_w_attribute(cur_ind) + '\n')
+        else:
+            f.write(cur_ind+'<' + self.tag + '>' + '\n')
+        for component in self.content:
+            try:
+                component.render(file_out, cur_ind+self.indent)
+            except AttributeError:
+                f.write(cur_ind+component)
+                f.write('\n')
+        f.write(cur_ind+'<\\'+self.tag+'>'+'\n')
+
+    def opening_tag_w_attribute(self, cur_ind):
+        tag_str = cur_ind+'<'+self.tag+' '+self.another_attrib_key+':'+'"'+self.another_attrib_value+'"'+'>'
+        return tag_str
+
+
+class Html(Element):
+    tag = "html"
+
+    def render(self, file_out, cur_ind=""):
+        f = file_out
+        f.write('<!DOCTYPE html>'+'\n')
+        Element.render(self, f)
```

Comment:
note that it is better to use `super().render(f)`

you can read more about this here: https://stackoverflow.com/questions/21639788/difference-between-super-and-calling-superclass-directly